src/inc/gcdecoder.cpp

```diff
@@ -86,19 +86,20 @@ size_t FASTCALL decodeSigned(PTR_CBYTE src, int* val)
 #pragma optimize("tgy", on)
 #endif
 
-PTR_CBYTE FASTCALL decodeHeader(PTR_CBYTE table, InfoHdr* header)
+PTR_CBYTE FASTCALL decodeHeader(PTR_CBYTE table, UINT32 version, InfoHdr* header)
 {
     LIMITED_METHOD_DAC_CONTRACT;
 
-    BYTE byte = *table++;
-    BYTE encoding = byte & 0x7f;
-    
+    BYTE nextByte = *table++;
+    BYTE encoding = nextByte & 0x7f;
+    const BYTE maskHaveMoreBytesBit = MORE_BYTES_TO_FOLLOW - 1;
```

Comment:
This value is confusing.
Instead of this why not define a second constant:
#define ENCODING_MASK 0x7f